stub.go

```diff
@@ -1,85 +1,56 @@
 package mocka
 
 import (
-	"fmt"
 	"reflect"
 	"sync"
 
 	"github.com/MonsantoCo/mocka/match"
-	"github.com/pkg/errors"
 )
 
 // variables used for unit testing
 var _cloneValue = cloneValue
 
-// Returner describes the functionality to update the return values to the stub
-type Returner interface {
-	Return(...interface{}) error
-}
-
-// GetCaller describes the functionality to get information for a specific call to the stub
-type GetCaller interface {
-	GetCalls() []Call
-	GetCall(int) Call
-	GetFirstCall() Call
-	GetSecondCall() Call
-	GetThirdCall() Call
-	CallCount() int
-	CalledOnce() bool
-	CalledTwice() bool
-	CalledThrice() bool
-}
-
-// OnCallReturner describes the functionality to update the return values itself of based on the call index
-type OnCallReturner interface {
-	OnCaller
-	Returner
-}
-
-// Stub describes the functionality related to the stub replacement of a function
-type Stub interface {
-	Returner
-	GetCaller
-	OnCaller
-	WithArgs(...interface{}) OnCallReturner
-	ExecOnCall(func([]interface{}))
-	Restore()
-}
-
-type mockFunction struct {
+// Stub represents the stub for a function
+type Stub struct {
 	lock sync.RWMutex
 
+	testReporter  TestReporter
 	originalFunc  interface{}
 	functionPtr   interface{}
 	outParameters []interface{}
-	calls         []call
-	customArgs    []*customArguments
-	onCalls       []*onCall
+	calls         []Call
+	customArgs    []*CustomArguments
+	onCalls       []*OnCall
 	execFunc      func([]interface{})
 }
 
-// newMockFunction creates a new mock function and overrides the implementation of the original function.
-func newMockFunction(originalFuncPtr interface{}, returnValues []interface{}) (*mockFunction, error) {
+// newStub creates a stub function and overrides the implementation of the original function.
+func newStub(testReporter TestReporter, originalFuncPtr interface{}, returnValues []interface{}) *Stub {
 	if originalFuncPtr == nil {
-		return nil, errors.New("mocka: expected the first argument to be a pointer to a function, but received a nil")
+		testReporter.Errorf("mocka: expected the first argument to be a pointer to a function, but received a nil")
```

Comment:
Need to update error message to be second param.

The same for the following error messages